### DISCUSSION TITLE
Support for other TPU profiles

### DISCUSSION THREAD

**Discussion by drinkmorewaters** [blakeblackshear/frigate#4929] [Ask A Question]
Hello, Google Coral TPU is an enigma, like many google products it simply lives off its own existence. Dare I say I for-see it's [end](https://killedbygoogle.com/) 

Is there support for other platforms that have TPU support? The Rockchip orange pi 5 for example has up to 6 TOPS and is ACTUALLY obtainable, these chips are going to become common in mini PC's, media servers, routers etc There is also, iCatch, Allwinner, Fullhan that all have TPU AI processing support.

[OrangePi 5](https://www.aliexpress.com/item/1005004942239582.html?spm=a2g0o.productlist.0.0.672a7064yf4Aqp&algo_pvid=cacab166-836b-4802-9c15-850f083f5e10&algo_exp_id=cacab166-836b-4802-9c15-850f083f5e10-12&pdp_ext_f=%7B%22sku_id%22%3A%2212000031099030945%22%7D&pdp_npi=2%40dis%21USD%21125.0%2175.0%21%21%21%21%21%402132f35216681304776078350e22f4%2112000031099030945%21sea&curPageLogUid=39CyqKrDlzve&gatewayAdapt=4itemAdapt)

[Rockchip SOC](https://en.t-firefly.com/product/industry/rocrk3588spc)

**Comment by NickM-27** [comment]
I'd suggest checking out 0.12 beta1 which added support for Intel iGPUs and nvidia gpus as detectors along with the coral

**Reply by drinkmorewaters** [reply]
> I'd suggest checking out 0.12 beta1 which added support for Intel iGPUs and nvidia gpus as detectors along with the coral

Great I will take a look, was hoping for ARM based TPU support as there are just so many manufacturers, specifically in Taiwan/China working on adding TPU support for edge AI, mostly to support AI based triggers in cameras/DVR's which is a good fit for Frigate. Some cost mere dollars for the SOC and have 1 TOPS etc

But I understand it's not easy to support them all as they all have different implementations.

**Reply by NickM-27** [reply]
The Intel iGPU support (via openvino) also includes support for the Intel ncs2

**Comment by NateMeyer** [comment]
The biggest problem with the RK3588 boards coming out right now is that the software support needs a little time to catch up.  I have a Rock 5B that I'll be playing with next.  The first challenge will be getting a stable OS and ffmpeg with hw encode/decode support.

**Reply by drinkmorewaters** [reply]
Great piece of kit, as anything new it just takes a little time for the software to catch up. It would surely make a very decent DVR for frigate, as it has 4K/8K encoding performance for H.264/5 and also has built in TPU for offloading detections. But it's just one of many SOC chips coming out of Asia now with AI based ASICs. When these manufacturers can make their SOC's without issue during this chip shortage, but Google can't, you gotta wonder.

**Comment by podarok** [comment]
https://github.com/blakeblackshear/frigate/pull/5733 added `armnn` detector, please test
